utils/nfc-mfultralight.c

```diff
@@ -376,13 +388,19 @@ write_card(bool write_otp, bool write_lock, bool write_uid)
     }
   }
 
+  // NTAG Dynamic Lock Bytes are in different locations for each type
   for (uint32_t page = uiSkippedPages; page < uiBlocks; page++) {
-    if ((page == 0x2) && (!write_lock)) {
+    if (((iNTAGType == NTAG_NONE && page == 0x2) || \
+         (iNTAGType == NTAG_213 && page == 0x28) || \
+         (iNTAGType == NTAG_215 && page == 0x82) || \
+         (iNTAGType == NTAG_216 && page == 0xe2)) && (!write_lock)) {
       printf("s");
       uiSkippedPages++;
       continue;
     }
-    if ((page == 0x3) && (!write_otp)) {
+    // NTAG doesn't have OTP blocks but Static Lock Bytes are OTP
+    if (((iNTAGType == NTAG_NONE && page == 0x3) || \
+        (iNTAGType && page == 0x2)) && (!write_otp)) {
```

Comment:
in (iNTAGType && page == 0x2)
please write iNTAGType != NTAG_NONE to not implicitly depend on NTAG_NONE value.

Reply:
OK. However, I've just realised that the behaviour of the Ultralight LOCK bytes is the same - they are also OTP (if you set them), so maybe my logic/phrasing is now rather confusing. The EV1 Ultralight also has a trailing Dynamic lock byte section... I think I have to do this differently! :) Stand by!!!